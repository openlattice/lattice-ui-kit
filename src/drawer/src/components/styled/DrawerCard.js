// @flow

import styled, { css } from 'styled-components';
import { WHITE } from '../../../../colors';

type Props = {
  side :'left' | 'right';
};

const getPosition = (props :Props) => {
  const { side } = props;
  switch (side) {
    case 'right':
      return css`
        right: 0;
      `;

    case 'left':
    default:
      return css`
        left: 0;
      `;
  }
};

const DrawerCard = styled.div`
  background-color: ${WHITE};
  box-shadow: 0 0 30px 0 rgba(0, 0, 0, 0.2);
  height: 100%;
  position: fixed;
  top: 0;
  width: 300px;
  z-index: 900;

  ${getPosition};
`;

export default DrawerCard;
