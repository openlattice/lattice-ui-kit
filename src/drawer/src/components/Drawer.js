// @flow

import React from 'react';
import type { Element } from 'react';

import Overlay from '../../../overlay';
import DrawerCard from './styled/DrawerCard';

type Props = {
  children ? :Node | Element<any>;
  isOpen ? :boolean;
  onClose ? :() => void;
  shouldCloseOnOutsideClick ? :boolean;
  side ? :'left' | 'right';
  transparentOverlay ? :boolean;
}

/*
 * Inspiration:
 * https://atlaskit.atlassian.com/packages/core/drawer
 * https://evergreen.surge.sh/components/side-sheet
 */
const Drawer = (props :Props) => {
  const {
    children,
    isOpen,
    onClose,
    shouldCloseOnOutsideClick,
    side,
    transparentOverlay,
  } = props;

  return (
    <Overlay
        onClose={onClose}
        isVisible={isOpen}
        isScrollable
        shouldCloseOnClick={shouldCloseOnOutsideClick}
        transparent={transparentOverlay}>
      <DrawerCard side={side}>
        {children}
      </DrawerCard>
    </Overlay>
  );
};

Drawer.defaultProps = {
  children: undefined,
  isOpen: false,
  onClose: undefined,
  shouldCloseOnOutsideClick: true,
  side: 'left',
  transparentOverlay: false,
};

export default Drawer;
