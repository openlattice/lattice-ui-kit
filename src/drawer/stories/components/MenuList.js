// @flow

import React from 'react';

import styled from 'styled-components';
import {
  faCommentDots,
  faFileAlt,
  faInbox,
  faSignOut,
  faUserCircle,
  faUsers,
} from '@fortawesome/free-solid-svg-icons';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

import { PURPLES } from '../../../colors';

const Icon = styled(FontAwesomeIcon)`
  margin-right: 20px;
`;

const MenuItemWrapper = styled.div`
  font-weight: 500;
  padding: 15px 30px;

  :hover {
    background-color: ${PURPLES[5]};
    color: ${PURPLES[1]};
    cursor: pointer;
  }
`;

const MenuListWrapper = styled.div`
  box-sizing: border-box;
  display: flex;
  flex-direction: column;
  height: 100%;
  padding: 15px 0;

  > ${MenuItemWrapper}:last-of-type {
    margin-top: auto;
  }
`;

const menuValues = [
  {
    icon: faUsers,
    text: 'People'
  },
  {
    icon: faFileAlt,
    text: 'New Report'
  },
  {
    icon: faInbox,
    text: 'Inbox'
  },
  {
    icon: faUserCircle,
    text: 'Profile'
  },
  {
    icon: faCommentDots,
    text: 'Feedback'
  },
  {
    icon: faSignOut,
    text: 'Sign Out'
  }
];

const MenuItem = ({ icon, text } :any) => (
  <MenuItemWrapper>
    <Icon icon={icon} fixedWidth />
    {text}
  </MenuItemWrapper>
);

const MenuList = () => (
  <MenuListWrapper>
    {
      menuValues.map((value) => {
        const { icon, text } = value;
        return <MenuItem key={text} icon={icon} text={text} />;
      })
    }
  </MenuListWrapper>
);

export default MenuList;
