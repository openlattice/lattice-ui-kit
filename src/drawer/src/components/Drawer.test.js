import React from 'react';
import toJson from 'enzyme-to-json';
import { mount } from 'enzyme';

import Drawer from './Drawer';
import DrawerCard from './styled/DrawerCard';
import MenuList from '../../stories/components/MenuList';

describe('Drawer', () => {

  const mockAddEventListenerMap = {};
  document.addEventListener = jest.fn((event, cb) => {
    mockAddEventListenerMap[event] = cb;
  });

  const mockRemoveEventListenerMap = {};
  document.removeEventListener = jest.fn((event, cb) => {
    mockRemoveEventListenerMap[event] = cb;
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  describe('snapshots', () => {
    test('isOpen = false', () => {
      const wrapper = mount(<Drawer />);
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    test('isOpen = true', () => {
      const wrapper = mount(<Drawer isOpen />);
      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });

  describe('props', () => {

    describe('isOpen', () => {
      test('should set Overlay isVisible to true when isOpen=true', () => {
        const wrapper = mount(<Drawer isOpen />);
        expect(wrapper.find('Overlay').prop('isVisible')).toEqual(true);
      });

      test('should set Overlay isVisible to false by default', () => {
        const wrapper = mount(<Drawer />);
        expect(wrapper.find('Overlay').prop('isVisible')).toEqual(false);
      });

      test('should set Overlay isVisible to false when isOpen=false', () => {
        const wrapper = mount(<Drawer isOpen={false} />);
        expect(wrapper.find('Overlay').prop('isVisible')).toEqual(false);
      });
    });

    describe('onClose', () => {

      test('should invoke onClose when clicking on overlay', () => {
        const mockOnClose = jest.fn();
        const wrapper = mount(<Drawer onClose={mockOnClose} isOpen />);

        const overlayWrapper = wrapper.find('Overlay');
        expect(overlayWrapper.prop('onClose')).toEqual(mockOnClose);

        expect(mockOnClose).toHaveBeenCalledTimes(0);
        overlayWrapper.find('OverlayInnerContainer').simulate('click');
        expect(mockOnClose).toHaveBeenCalledTimes(1);
      });
    });

    describe('shouldCloseOnOutsideClick', () => {
      test('should not invoke onClose when shouldCloseOnOutsideClick=true', () => {
        const mockOnClose = jest.fn();
        const wrapper = mount(
          <Drawer
              onClose={mockOnClose}
              isOpen
              shouldCloseOnOutsideClick={false} />
        );

        const overlayWrapper = wrapper.find('Overlay');
        expect(overlayWrapper.prop('onClose')).toEqual(mockOnClose);

        expect(mockOnClose).toHaveBeenCalledTimes(0);
        overlayWrapper.find('OverlayInnerContainer').simulate('click');
        expect(mockOnClose).toHaveBeenCalledTimes(0);
      });
    });

    describe('side', () => {

      test('should set to "left" by default', () => {
        const wrapper = mount(<Drawer isOpen />);
        const drawerCardWrapper = wrapper.find('DrawerCard');

        expect(drawerCardWrapper.prop('side')).toEqual('left');
        expect(drawerCardWrapper).toHaveStyleRule('left', '0');
      });

      test('should render DrawerCard on right when side="right"', () => {
        const wrapper = mount(<Drawer isOpen side="right" />);
        const drawerCardWrapper = wrapper.find('DrawerCard');

        expect(drawerCardWrapper.prop('side')).toEqual('right');
        expect(drawerCardWrapper).toHaveStyleRule('right', '0');
      });

    });

    describe('shouldCloseOnEscape', () => {
      test('shouldCloseOnEscape should be true by default', () => {
        const mockOnClose = jest.fn();
        const wrapper = mount(
          <DrawerCard onClose={mockOnClose} />
        );

        expect(wrapper.find('DrawerCard')).toHaveLength(1);
        expect(mockOnClose).toHaveBeenCalledTimes(0);
        expect(mockAddEventListenerMap).toHaveProperty('keydown');
      });

      test('should add and remove event handlers when Drawer is toggled', () => {
        const mockOnClose = jest.fn();
        jest.useFakeTimers();
        const wrapper = mount(
          <Drawer
              onClose={mockOnClose}
              isOpen
              shouldCloseOnEscape />
        );

        expect(wrapper.find('DrawerCard')).toHaveLength(1);
        expect(mockOnClose).toHaveBeenCalledTimes(0);
        expect(mockAddEventListenerMap).toHaveProperty('keydown');
        expect(mockRemoveEventListenerMap).not.toHaveProperty('keydown');
        wrapper.setProps({ isOpen: false });
        jest.runAllTimers();
        expect(mockRemoveEventListenerMap).toHaveProperty('keydown');
      });

      test('should invoke onClose on escape keydown when true', () => {
        const mockOnClose = jest.fn();
        const wrapper = mount(
          <Drawer
              onClose={mockOnClose}
              isOpen
              shouldCloseOnEscape />
        );

        expect(wrapper.find('DrawerCard')).toHaveLength(1);
        expect(mockOnClose).toHaveBeenCalledTimes(0);
        mockAddEventListenerMap.keydown({ key: 'Escape', code: 'Escape' });
        expect(mockOnClose).toHaveBeenCalledTimes(1);
      });

      test('should not invoke onClose on misc keydown', () => {
        const mockOnClose = jest.fn();
        const wrapper = mount(
          <Drawer
              onClose={mockOnClose}
              isOpen
              shouldCloseOnEscape />
        );

        expect(wrapper.find('DrawerCard')).toHaveLength(1);
        expect(mockOnClose).toHaveBeenCalledTimes(0);
        mockAddEventListenerMap.keydown({ key: 'A', code: 'A' });
        expect(mockOnClose).toHaveBeenCalledTimes(0);
      });

      test('should not invoke onClose on escape keydown when false', () => {
        const mockOnClose = jest.fn();
        const wrapper = mount(
          <Drawer
              onClose={mockOnClose}
              isOpen
              shouldCloseOnEscape={false} />
        );

        expect(wrapper.find('DrawerCard')).toHaveLength(1);
        expect(mockOnClose).toHaveBeenCalledTimes(0);
        mockAddEventListenerMap.keydown({ key: 'Escape', code: 'Escape' });
        expect(mockOnClose).toHaveBeenCalledTimes(0);
      });
    });

    describe('children', () => {
      test('should render children inside DrawerCard', () => {
        const wrapper = mount(
          <Drawer isOpen>
            <MenuList />
          </Drawer>
        );

        const drawerCardWrapper = wrapper.find('DrawerCard');
        expect(drawerCardWrapper.contains(MenuList)).toEqual(true);
      });
    });

  });

  describe('render', () => {
    test('should render with styles', () => {
      const wrapper = mount(<Drawer isOpen />);

      const drawerCardWrapper = wrapper.find('DrawerCard');
      expect(drawerCardWrapper).toHaveStyleRule('position', 'fixed');
      expect(drawerCardWrapper).toHaveStyleRule('height', '100%');
      expect(drawerCardWrapper).toHaveStyleRule('z-index', '1000');

    });
  });

});
