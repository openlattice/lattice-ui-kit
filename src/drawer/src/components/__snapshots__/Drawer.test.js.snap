// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Drawer snapshots isOpen = false 1`] = `
<Drawer
  isOpen={false}
  shouldCloseOnEscape={true}
  shouldCloseOnOutsideClick={true}
  side="left"
  transparentOverlay={false}
>
  <Overlay
    isScrollable={true}
    isVisible={false}
    shouldCloseOnClick={true}
    transparent={false}
  >
    <CSSTransition
      classNames="luk-fade"
      in={false}
      mountOnEnter={true}
      timeout={200}
      unmountOnExit={true}
    >
      <Transition
        appear={false}
        enter={true}
        exit={true}
        in={false}
        mountOnEnter={true}
        onEnter={[Function]}
        onEntered={[Function]}
        onEntering={[Function]}
        onExit={[Function]}
        onExited={[Function]}
        onExiting={[Function]}
        timeout={200}
        unmountOnExit={true}
      />
    </CSSTransition>
  </Overlay>
  <CSSTransition
    classNames="luk-slide"
    in={false}
    mountOnEnter={true}
    timeout={200}
    unmountOnExit={true}
  >
    <Transition
      appear={false}
      enter={true}
      exit={true}
      in={false}
      mountOnEnter={true}
      onEnter={[Function]}
      onEntered={[Function]}
      onEntering={[Function]}
      onExit={[Function]}
      onExited={[Function]}
      onExiting={[Function]}
      timeout={200}
      unmountOnExit={true}
    />
  </CSSTransition>
</Drawer>
`;

exports[`Drawer snapshots isOpen = true 1`] = `
.c0 {
  background-color: rgba(8,0,25,0.3);
  bottom: 0;
  display: block;
  left: 0;
  position: fixed;
  right: 0;
  top: 0;
  z-index: 1000;
}

.c0.luk-fade-appear {
  opacity: 0;
}

.c0.luk-fade-appear-active {
  opacity: 1;
  -webkit-transition: opacity 200ms ease-out;
  transition: opacity 200ms ease-out;
}

.c0.luk-fade-enter {
  opacity: 0;
}

.c0.luk-fade-enter-active {
  opacity: 1;
  -webkit-transition: opacity 200ms ease-out;
  transition: opacity 200ms ease-out;
}

.c0.luk-fade-exit {
  opacity: 1;
}

.c0.luk-fade-exit-active {
  opacity: 0;
  -webkit-transition: opacity 200ms ease-out;
  transition: opacity 200ms ease-out;
}

.c1 {
  display: block;
  height: 100%;
  position: absolute;
  overflow-y: scroll;
  width: 100%;
}

.c2 {
  background-color: #ffffff;
  box-shadow: 0 0 30px 0 rgba(0,0,0,0.2);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 100%;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  min-width: 300px;
  overflow-y: auto;
  position: fixed;
  top: 0;
  z-index: 1000;
  left: 0;
  --transform: translate(-100%);
}

.c2.luk-slide-appear {
  -webkit-transform: var(--transform);
  -ms-transform: var(--transform);
  transform: var(--transform);
}

.c2.luk-slide-appear-active {
  -webkit-transform: translate(0);
  -ms-transform: translate(0);
  transform: translate(0);
  -webkit-transition: -webkit-transform 250ms ease-out;
  -webkit-transition: transform 250ms ease-out;
  transition: transform 250ms ease-out;
}

.c2.luk-slide-enter {
  -webkit-transform: var(--transform);
  -ms-transform: var(--transform);
  transform: var(--transform);
}

.c2.luk-slide-enter-active {
  -webkit-transform: translate(0);
  -ms-transform: translate(0);
  transform: translate(0);
  -webkit-transition: -webkit-transform 250ms ease-out;
  -webkit-transition: transform 250ms ease-out;
  transition: transform 250ms ease-out;
}

.c2.luk-slide-exit {
  -webkit-transform: translate(0);
  -ms-transform: translate(0);
  transform: translate(0);
}

.c2.luk-slide-exit-active {
  -webkit-transform: var(--transform);
  -ms-transform: var(--transform);
  transform: var(--transform);
  -webkit-transition: -webkit-transform 200ms ease-out;
  -webkit-transition: transform 200ms ease-out;
  transition: transform 200ms ease-out;
}

<Drawer
  isOpen={true}
  shouldCloseOnEscape={true}
  shouldCloseOnOutsideClick={true}
  side="left"
  transparentOverlay={false}
>
  <Overlay
    isScrollable={true}
    isVisible={true}
    shouldCloseOnClick={true}
    transparent={false}
  >
    <CSSTransition
      classNames="luk-fade"
      in={true}
      mountOnEnter={true}
      timeout={200}
      unmountOnExit={true}
    >
      <Transition
        appear={false}
        enter={true}
        exit={true}
        in={true}
        mountOnEnter={true}
        onEnter={[Function]}
        onEntered={[Function]}
        onEntering={[Function]}
        onExit={[Function]}
        onExited={[Function]}
        onExiting={[Function]}
        timeout={200}
        unmountOnExit={true}
      >
        <Portal>
          <Portal
            containerInfo={
              .c0 {
  background-color: rgba(8,0,25,0.3);
  bottom: 0;
  display: block;
  left: 0;
  position: fixed;
  right: 0;
  top: 0;
  z-index: 1000;
}

.c0.luk-fade-appear {
  opacity: 0;
}

.c0.luk-fade-appear-active {
  opacity: 1;
  -webkit-transition: opacity 200ms ease-out;
  transition: opacity 200ms ease-out;
}

.c0.luk-fade-enter {
  opacity: 0;
}

.c0.luk-fade-enter-active {
  opacity: 1;
  -webkit-transition: opacity 200ms ease-out;
  transition: opacity 200ms ease-out;
}

.c0.luk-fade-exit {
  opacity: 1;
}

.c0.luk-fade-exit-active {
  opacity: 0;
  -webkit-transition: opacity 200ms ease-out;
  transition: opacity 200ms ease-out;
}

.c1 {
  display: block;
  height: 100%;
  position: absolute;
  overflow-y: scroll;
  width: 100%;
}

<div
                lattice-portal-inner-container=""
              >
                <div
                  class="c0"
                >
                  <div
                    class="c1"
                  />
                </div>
              </div>
            }
          >
            <LockToggle
              accountForScrollbars={true}
              isActive={true}
            >
              <SheetProvider
                accountForScrollbars={true}
                isActive={true}
              >
                <TouchProvider
                  accountForScrollbars={true}
                  isActive={true}
                >
                  <ScrollLock
                    accountForScrollbars={true}
                    isActive={true}
                  >
                    <TouchScrollable>
                      <NodeResolver
                        innerRef={[Function]}
                      >
                        <OverlayOuterContainer
                          transparent={false}
                        >
                          <StyledComponent
                            forwardedComponent={
                              Object {
                                "$$typeof": Symbol(react.forward_ref),
                                "attrs": Array [],
                                "componentStyle": ComponentStyle {
                                  "componentId": "StyledOverlayComponents__OverlayOuterContainer-ol4tzw-0",
                                  "isStatic": false,
                                  "lastClassName": "c0",
                                  "rules": Array [
                                    "background-color:",
                                    [Function],
                                    ";bottom:0;display:block;left:0;position:fixed;right:0;top:0;z-index:1000;",
                                    "&.luk-fade-appear{opacity:0;};&.luk-fade-appear-active{opacity:1;transition:opacity ",
                                    "200ms",
                                    " ease-out;};&.luk-fade-enter{opacity:0;};&.luk-fade-enter-active{opacity:1;transition:opacity ",
                                    "200ms",
                                    " ease-out;};&.luk-fade-exit{opacity:1;};&.luk-fade-exit-active{opacity:0;transition:opacity ",
                                    "200ms",
                                    " ease-out;};",
                                    ";",
                                  ],
                                },
                                "displayName": "OverlayOuterContainer",
                                "foldedComponentIds": Array [],
                                "render": [Function],
                                "styledComponentId": "StyledOverlayComponents__OverlayOuterContainer-ol4tzw-0",
                                "target": "div",
                                "toString": [Function],
                                "warnTooManyClasses": [Function],
                                "withComponent": [Function],
                              }
                            }
                            forwardedRef={null}
                            transparent={false}
                          >
                            <div
                              className="c0"
                            >
                              <OverlayInnerContainer
                                isScrollable={true}
                                onClick={[Function]}
                              >
                                <StyledComponent
                                  forwardedComponent={
                                    Object {
                                      "$$typeof": Symbol(react.forward_ref),
                                      "attrs": Array [],
                                      "componentStyle": ComponentStyle {
                                        "componentId": "StyledOverlayComponents__OverlayInnerContainer-ol4tzw-1",
                                        "isStatic": false,
                                        "lastClassName": "c1",
                                        "rules": Array [
                                          "display:block;height:100%;position:absolute;overflow-y:",
                                          [Function],
                                          ";width:100%;",
                                        ],
                                      },
                                      "displayName": "OverlayInnerContainer",
                                      "foldedComponentIds": Array [],
                                      "render": [Function],
                                      "styledComponentId": "StyledOverlayComponents__OverlayInnerContainer-ol4tzw-1",
                                      "target": "div",
                                      "toString": [Function],
                                      "warnTooManyClasses": [Function],
                                      "withComponent": [Function],
                                    }
                                  }
                                  forwardedRef={null}
                                  isScrollable={true}
                                  onClick={[Function]}
                                >
                                  <div
                                    className="c1"
                                    onClick={[Function]}
                                  />
                                </StyledComponent>
                              </OverlayInnerContainer>
                            </div>
                          </StyledComponent>
                        </OverlayOuterContainer>
                      </NodeResolver>
                    </TouchScrollable>
                  </ScrollLock>
                </TouchProvider>
              </SheetProvider>
            </LockToggle>
          </Portal>
        </Portal>
      </Transition>
    </CSSTransition>
  </Overlay>
  <CSSTransition
    classNames="luk-slide"
    in={true}
    mountOnEnter={true}
    timeout={200}
    unmountOnExit={true}
  >
    <Transition
      appear={false}
      enter={true}
      exit={true}
      in={true}
      mountOnEnter={true}
      onEnter={[Function]}
      onEntered={[Function]}
      onEntering={[Function]}
      onExit={[Function]}
      onExited={[Function]}
      onExiting={[Function]}
      timeout={200}
      unmountOnExit={true}
    >
      <Portal>
        <Portal
          containerInfo={
            .c0 {
  background-color: #ffffff;
  box-shadow: 0 0 30px 0 rgba(0,0,0,0.2);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 100%;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  min-width: 300px;
  overflow-y: auto;
  position: fixed;
  top: 0;
  z-index: 1000;
  left: 0;
  --transform: translate(-100%);
}

.c0.luk-slide-appear {
  -webkit-transform: var(--transform);
  -ms-transform: var(--transform);
  transform: var(--transform);
}

.c0.luk-slide-appear-active {
  -webkit-transform: translate(0);
  -ms-transform: translate(0);
  transform: translate(0);
  -webkit-transition: -webkit-transform 250ms ease-out;
  -webkit-transition: transform 250ms ease-out;
  transition: transform 250ms ease-out;
}

.c0.luk-slide-enter {
  -webkit-transform: var(--transform);
  -ms-transform: var(--transform);
  transform: var(--transform);
}

.c0.luk-slide-enter-active {
  -webkit-transform: translate(0);
  -ms-transform: translate(0);
  transform: translate(0);
  -webkit-transition: -webkit-transform 250ms ease-out;
  -webkit-transition: transform 250ms ease-out;
  transition: transform 250ms ease-out;
}

.c0.luk-slide-exit {
  -webkit-transform: translate(0);
  -ms-transform: translate(0);
  transform: translate(0);
}

.c0.luk-slide-exit-active {
  -webkit-transform: var(--transform);
  -ms-transform: var(--transform);
  transform: var(--transform);
  -webkit-transition: -webkit-transform 200ms ease-out;
  -webkit-transition: transform 200ms ease-out;
  transition: transform 200ms ease-out;
}

<div
              lattice-portal-inner-container=""
            >
              <div
                class="c0"
              />
            </div>
          }
        >
          <DrawerCard
            shouldCloseOnEscape={true}
            side="left"
          >
            <DrawerCard__DrawerCardWrapper
              side="left"
            >
              <StyledComponent
                forwardedComponent={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "attrs": Array [],
                    "componentStyle": ComponentStyle {
                      "componentId": "DrawerCard__DrawerCardWrapper-sc-1ka5kr5-0",
                      "isStatic": false,
                      "lastClassName": "c2",
                      "rules": Array [
                        "background-color:",
                        "#ffffff",
                        ";box-shadow:0 0 30px 0 rgba(0,0,0,0.2);display:flex;flex-direction:column;height:100%;justify-content:flex-start;min-width:300px;overflow-y:auto;position:fixed;top:0;z-index:1000;",
                        [Function],
                        ";",
                        [Function],
                        ";&.luk-slide-appear{transform:var(--transform);};&.luk-slide-appear-active{transform:translate(0);transition:transform ",
                        "250ms",
                        " ease-out;};&.luk-slide-enter{transform:var(--transform);};&.luk-slide-enter-active{transform:translate(0);transition:transform ",
                        "250ms",
                        " ease-out;};&.luk-slide-exit{transform:translate(0);};&.luk-slide-exit-active{transform:var(--transform);transition:transform ",
                        "200ms",
                        " ease-out;}",
                        ";",
                      ],
                    },
                    "displayName": "DrawerCard__DrawerCardWrapper",
                    "foldedComponentIds": Array [],
                    "render": [Function],
                    "styledComponentId": "DrawerCard__DrawerCardWrapper-sc-1ka5kr5-0",
                    "target": "div",
                    "toString": [Function],
                    "warnTooManyClasses": [Function],
                    "withComponent": [Function],
                  }
                }
                forwardedRef={null}
                side="left"
              >
                <div
                  className="c2"
                />
              </StyledComponent>
            </DrawerCard__DrawerCardWrapper>
          </DrawerCard>
        </Portal>
      </Portal>
    </Transition>
  </CSSTransition>
</Drawer>
`;
